cmd/contact-auditor/main.go

```diff
@@ -0,0 +1,203 @@
+package main
+
+import (
+	"database/sql"
+	"encoding/json"
+	"errors"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"strings"
+	"time"
+
+	"github.com/jmhodges/clock"
+	"github.com/letsencrypt/boulder/cmd"
+	blog "github.com/letsencrypt/boulder/log"
+	"github.com/letsencrypt/boulder/policy"
+)
+
+var queryInterrupted bool
+
+type contactAuditor struct {
+	db     *sql.DB
+	logger blog.Logger
+	clk    clock.Clock
+	grace  time.Duration
+}
+
+type result struct {
+	// Receiver for the `id` column.
+	ID int64
+
+	// Receiver for the `contact` column.
+	Contact  []byte
+	contacts []string
```

Comment:
It's a bit odd to have a single struct that acts as the receiver for scanning, and also has a field that is _not_ populated by Scan.

The way I would do this is: when calling Scan, read the values into local variables. Then parse the single `contact []byte` string into multiple contacts, and make a:

```
type result struct {
  id int64
  contacts []string
}
```

Reply:
Yeah, this is totally fair. I've gone ahead and refactored to this approach.